src/python/pants/task/task.py

```diff
@@ -333,14 +333,19 @@ def invalidated(self,
       self.fingerprint)
     cache_manager = InvalidationCacheManager(self.workdir,
                                              cache_key_generator,
-                                             self._build_invalidator_dir,
+                                             self._build_invalidator(),
                                              invalidate_dependents,
                                              fingerprint_strategy=fingerprint_strategy,
                                              invalidation_report=self.context.invalidation_report,
                                              task_name=type(self).__name__,
                                              task_version=self.implementation_version_str(),
                                              artifact_write_callback=self.maybe_write_artifact)
 
+    # If this Task's execution has been forced, invalidate all our target fingerprints.
+    if self._cache_factory.ignore and not self._force_invalidated:
+      self.invalidate()
```

Comment:
I think this is where the tangle I mentioned between `BuildInvalidator`, `InvalidationCacheManager` and `CacheFactory` comes in. Without solving that, `Task` has to coordinate for now. I'd love to not solve that tangle at a first pass though. What do you say to an issue tracking cleaning that triangle up and thus cleaning this code up?